api/tokens.go

```diff
@@ -71,6 +74,54 @@ func TokensHandler() http.HandlerFunc {
 	}
 }
 
+func (api *API) SignOutHandler(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		ctx := req.Context()
+		var errorList []apierrors.IndividualError
+		field := ""
+		param := ""
+
+		authString := req.Header.Get("Authorization")
+		if authString == "" {
+			invalidTokenErrorBody := apierrors.IndividualErrorBuilder(invalidTokenError, invalidTokenMessage, field, param)
+			errorList = append(errorList, invalidTokenErrorBody)
+			log.Event(ctx, "no authorization header provided", log.ERROR)
+			errorResponseBody := apierrors.ErrorResponseBodyBuilder(errorList)
+			writeErrorResponse(ctx, w, http.StatusBadRequest, errorResponseBody)
```

Comment:
This is interesting; I'm appending all validation errors to the an `errorList []models.IndividualError` array and then returning them all rather than individually (as the ticket requires this) by doing:

`		// report validation errors in response
		if len(errorList) != 0 {
			apierrors.WriteErrorResponse(ctx, w, http.StatusBadRequest, apierrors.ErrorResponseBodyBuilder(errorList))
			return
		}`

Reply:
I think thats the right approach in general.
But in this case all the validation checks are on the same piece of input and if one fails all subsequent ones can't pass, and in some cases would cause panic states (e.g. trying to access the 2nd element after the split when the header was malformed and doesn't have a second element).
So I felt it was better to return after a validation failure rather than get in to a lot of nested ifs.